Program/src/port.c

```diff
@@ -0,0 +1,303 @@
+/*------------------------------------------------------------------------------
+This file is protected under copyright. If you want to use it,
+please include this text, that is my only stipulation.  
+
+Author: Evan Giese
+------------------------------------------------------------------------------*/
+#include "port.h"
+
+
+#ifdef POSIX_PORT
+    #include <pthread.h>
+    #include <errno.h>
+    #include <limits.h>
+    #include <stdlib.h>
+    #include <stdio.h>
+    #include <arpa/inet.h>
+    #include <stdarg.h>
+#endif
+
+#ifdef POSIX_FILESYSTEM
+    #include <unistd.h>
+    #include <fcntl.h>
+#endif
+
+#ifdef FREE_RTOS_PORT 
+    #include "FreeRTOS.h"
+    #include "task.h"
+    #include "portable.h"
+#endif
+
+#ifdef CSP_NETWORK
+    #include "csp.h"
+#endif
+
+
+/*------------------------------------------------------------------------------
+    File system port functions, these are used to interchange different 
+    File systems
+------------------------------------------------------------------------------*/
+int ssp_rename(const char *old, const char *new) {
+    #ifdef POSIX_FILESYSTEM
+        return rename(old, new);
+    #endif
+}
+
+int ssp_write(int fd, const void *buf, size_t count) {
+    #ifdef POSIX_FILESYSTEM
+        return write(fd, buf, count);
+    #endif
+}
+
+
+int ssp_read(int fd, char* buff, size_t size) {
+    #ifdef POSIX_FILESYSTEM
+        return read(fd, buff, size);
+    #endif
+
+}
+
+//SEEK_END 2  SEEK_CUR 1  SEEK_SET 0 
+int ssp_lseek(int fd, int offset, int whence) {
+    #ifdef POSIX_FILESYSTEM
+        return lseek(fd, offset, whence);
+    #endif
+} 
+
+int ssp_open(char *pathname, int flags) {
+    #ifdef POSIX_FILESYSTEM
+        //open with read and write permissions
+        return open(pathname, flags, 0666);
+    #endif
+}
+
+int ssp_close(int fd) {
+    #ifdef POSIX_FILESYSTEM
+        return close(fd);
+    #endif
+}
+
+
+/*------------------------------------------------------------------------------
+    Network port functions, these are used to interchange different network
+    stacks
+------------------------------------------------------------------------------*/
+
+void ssp_sendto(Response res) {
```

Comment:
mmm i don't know. The program will continue to run even if there is errors here. It will try re-sending any missed packets. If there are errors for some reason in the send library, i'm not sure what I can do as an application writer. Kind of just brainstorming, maybe if there are too many errors I should stop the program entirely? -- I need to look into this.